crates/oxc_regular_expression/src/parser/reader/template_literal_parser/parser_impl.rs

```diff
@@ -0,0 +1,573 @@
+                        if self.eat('}') {
+                            return Ok(Some(hex_digits));
+                        }
+
+                        // Missing closing '}'
+                        return Err(diagnostics::invalid_unicode_escape(Span::new(
+                            self.options.span_offset + offset_start,
+                            self.options.span_offset + self.offset(),
+                        )));
+                    }
+                    Ok(_) => {
+                        // No valid hex digits or out of range
+                        return Err(diagnostics::invalid_unicode_escape(Span::new(
+                            self.options.span_offset + offset_start,
+                            self.options.span_offset + self.offset(),
+                        )));
+                    }
+                    Err(e) => return Err(e),
+                }
+            }
+            self.rewind(checkpoint);
+        }
+
+        Ok(None)
+    }
+
+    // ```
+    // LineContinuation ::
+    //   \ LineTerminatorSequence
+    //
+    // ```
+    fn parse_line_continuation(&mut self) -> Option<u32> {
+        let checkpoint = self.checkpoint();
+
+        if self.eat('\\') {
+            if let Some(terminator) = self.parse_line_terminator_sequence() {
+                return Some(terminator);
+            }
+        }
+
+        self.rewind(checkpoint);
+        None
+    }
+
+    // ```
+    // LineTerminatorSequence ::
+    //   <LF>
+    //   <CR> [lookahead ≠ <LF>]
+    //   <LS>
+    //   <PS>
+    //   <CR> <LF>
+    // ```
+    fn parse_line_terminator_sequence(&mut self) -> Option<u32> {
+        let checkpoint = self.checkpoint();
+
+        if self.peek() == Some(LF) {
+            self.advance();
+            return Some(LF as u32);
+        }
+        if self.peek() == Some(CR) && self.peek2() != Some(LF) {
+            self.advance();
+            return Some(CR as u32);
+        }
+        if self.peek() == Some(LS) {
+            self.advance();
+            return Some(LS as u32);
+        }
+        if self.peek() == Some(PS) {
+            self.advance();
+            return Some(PS as u32);
+        }
+        // NOTE: CR+LF can not represent as a single code point.
+        // I don't know the best way to handle this.
+        // To distinguish this from CR and LF, structural change is needed...
+        if self.peek() == Some(CR) && self.peek2() == Some(LF) {
+            self.advance();
+            self.advance();
+            // NOTE: CR+LF can not represent as a single code point.
+            // I don't know the best way to handle this.
+            // To distinguish this from CR and LF, structural change is needed...
+            return Some(LF as u32);
+        }
+
+        self.rewind(checkpoint);
+        None
+    }
+
+    // ```
+    // LineTerminator ::
+    //   <LF>
+    //   <CR>
+    //   <LS>
+    //   <PS>
+    // ```
+    fn parse_line_terminator(&mut self) -> Option<u32> {
```

Comment:
You can use `character::is_line_terminator` and inline it like https://github.com/oxc-project/oxc/blob/ca4f52fa62af96b72aec0c9ab8975ceb9d5b84bb/crates/oxc_regular_expression/src/parser/reader/string_literal_parser/parser_impl.rs#L153

Reply:
OK, my question was about usage at L156.

How about writing it like this?

```rs
if ch == '\\' || ch == '`' || is_line_terminator(ch) {
```

`self.parse_line_terminator().is_some()` should not call `advance()` for checking `SourceCharacter but not one of ` or \ or $ or LineTerminator`?